app/src/main/java/com/strv/movies/MainViewModel.kt

```diff
@@ -2,28 +2,28 @@ package com.strv.movies
 
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.strv.movies.data.database.MoviesDataStore
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
-import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
 class MainViewModel @AssistedInject constructor(
-    @Assisted savedStateHandle: SavedStateHandle,
-    @Assisted isDarkTheme: Boolean,
-): ViewModel() {
+    private val moviesDataStore: MoviesDataStore,
+    @Assisted savedStateHandle: SavedStateHandle
```

Comment:
We can now remove assisted inject. SavedStateHandle can be injected automatically by hilt when annotated with `@HiltViewModel`
Unless we would want to demonstrate assisted inject on it on my lesson, but I don't think this is the best use case for it

Reply:
👍